File: youkon/YouKon/src/main/java/com/dcsim/youkon/android/views/SubtractAlert.kt
package com.dcsim.youkon.android.views

import androidx.compose.material.AlertDialog
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.TextButton
import androidx.compose.runtime.Composable

/// Dialog that is shown when the user taps the `X` button to the left of a project or measurement
@Composable
fun SubtractAlert(
    title: String,
    confirmAction: () -> Unit,
    cancelAction: () -> Unit
) {
    AlertDialog(
        title = { Text("Delete $title") },
        text = { Text("Are you sure?") },
        onDismissRequest = { cancelAction() },
        confirmButton = {
            TextButton(onClick = {
                confirmAction()
            }) {
                Text("Delete",
                    color = MaterialTheme.colors.error
                )
            }
        },
        dismissButton = {
            TextButton(onClick = {
                cancelAction()
            }) {
                Text("Cancel")
            }
        }
    )
}

File: YouKonCMP/iosApp/iosApp/view/SubtractAlert.swift
//
//  SubtractAlert.swift
//  YouKon
//
//  Created by Eliott Radcliffe on 9/26/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI

/// Dialog that is shown when the user taps the `X` button to the left of a project or measurement
func SubtractAlert(
    title: String,
    confirmAction: @escaping () -> Void,
    cancelAction: @escaping () -> Void
) -> Alert {
    return Alert(
        title: Text("Delete \(title)"),
        message: Text("Are you sure?"),
        primaryButton: .destructive(Text("Delete")) {
            confirmAction()
        },
        secondaryButton: .cancel() {
            cancelAction()
        }
    )
}
